ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/StateManager.java

```diff
@@ -641,12 +642,12 @@ protected void packHeader (OutputObjectState os, Header hdr)
 
         Uid txId = ((hdr == null) ? null : hdr.getTxId());
         Uid processUid = ((hdr == null) ? null : hdr.getProcessId());
-        
+
         try
         {
             // pack the marker first.
 
-            os.packString(StateManager.marker);
+            os.packStringBytes(markerBytes);
```

Comment:
What is the performance difference between packing the bytes directly versus packing the string - is it that the string copy (`os.packString(StateManager.marker)`) is avoided?

Reply:
The calculation that transforms the String's internal representation (char[]) to the encoded form (byte[]) need not be repeated for every single StateManager instance (which is MANY), in addition to which you avoid the gc churn and copies to/from the short-lived byte[] that's needed by the encoder. The actual packing once you have the byte[] is then the same either way. The encoding is a trivial overhead on tx that hit the network, but worth a 1-2% on in-memory cases. May not have bothered, except it's easy to piggyback the optimization onto the bug fix.

Reply:
maybe a matter of taste but would not be better to use `StateManager.markerBytes` when static variable is accessed here?